src/type_tag.rs

```diff
@@ -12,7 +12,9 @@ pub enum TypeTag {
     // Experimental; may make no sense at runtime, as we will likely be unable to take the value of a macro
     Macro,
     String,
-    Nil,
+    Integer,
+    ISeq,
```

Comment:
ISeq here is an interface (or rather, it will be considered a protocol),  and I haven't decided how these will be represented -- I might just make a new TypeTag 

```rust
enum TypeTag { 
  ...
 Protocol(String),
 ...
```
.  You'll notice there's already a similar `IFn` TypeTag, but that's actually now a misnomer, as early on `IFn` was something else -- that `IFn` is basically a `Fn` now